### PR TITLE
[PPD-70] retrieve EC list

### PR DESCRIPTION
**Pull request by pasqualespica**
The purpose of this PR is to implements all functions contained in to `reporting-batch` AZ fn-app

image

#### List of Changes

Add : 
 -  mock `gpd` and `nodo` serivice
 - `TimerTrigger` function `ReportingBatchTrigger`

Upd : 
 - junit test
 - `QueueTrigger` function `RetrieveOrganizationsTrigger`

#### How Has This Been Tested?

- junit

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: reporting-batch/src/main/resources/wsdl/NodoPerPa.wsdl [hunk 1]
```diff
@@ -866,8 +866,7 @@
   </wsdl:binding>
   <wsdl:service name="PagamentiTelematiciRPTservice">
     <wsdl:port name="PagamentiTelematiciRPTPort" binding="tns:PagamentiTelematiciRPTbinding">
-      <soap:address location="https://gad.test.pagopa.gov.it/openspcoop2/proxy/PA/RPT6T/" />
-
+      <soap:address location="http://localhost:8086/nodo-per-pa/v1" />
```
Comment (jacopocarlini) [inline reporting-batch/src/main/resources/wsdl/NodoPerPa.wsdl:869]: is it okay to leave localhost?

File: reporting-batch/src/main/java/it/gov/pagopa/reporting/service/FlowsService.java [hunk 2]
```diff
@@ -40,8 +40,37 @@ public FlowsService(String storageConnectionString, String flowsTable, String fl
         this.logger = logger;
     }
 
+//    private void createTable() throws URISyntaxException, InvalidKeyException, StorageException {
```
Comment (jacopocarlini) [inline reporting-batch/src/main/java/it/gov/pagopa/reporting/service/FlowsService.java:43]: it is ok, but commented code is a code smell for sonar

File: reporting-batch/src/main/java/it/gov/pagopa/reporting/service/FlowsService.java [hunk 3]
```diff
@@ -111,14 +140,13 @@ public void flowsBatchProcessing(List<TipoIdRendicontazione> partition, String i
...
 
-        this.logger.log(Level.INFO, () -> "[FlowsService] ### 1");
-
         TableBatchOperation batchOperation = new TableBatchOperation();
 
         SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
 
-        partition.stream().forEach(flow -> batchOperation.insert(new FlowEntity(flow.getIdentificativoFlusso(),
-                sdf.format(flow.getDataOraFlusso().toGregorianCalendar().getTime()), idPA)));
+        partition.stream().forEach(flow -> batchOperation.insert(
```
Comment (jacopocarlini) [inline reporting-batch/src/main/java/it/gov/pagopa/reporting/service/FlowsService.java:147]:
```suggestion
        partition.forEach(flow -> batchOperation.insert(
```

File: reporting-batch/src/main/java/it/gov/pagopa/reporting/entity/OrganizationEntity.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public OrganizationEntity(String organizationId, String organizationOnboardingDate) {
+        this.partitionKey = organizationKey;
+        this.rowKey = organizationId;
+        this.organizationOnboardingDate = organizationOnboardingDate;
+    }
+
+    public OrganizationEntity(String organizationId) {
+        this.partitionKey = organizationKey;
+        this.rowKey = organizationId;
+        this.etag = "*";
```
Comment (jacopocarlini) [inline reporting-batch/src/main/java/it/gov/pagopa/reporting/entity/OrganizationEntity.java:26]: add a comment explaining why

File: reporting-batch/src/main/java/it/gov/pagopa/reporting/RetrieveOrganizations.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * This function will be invoked periodically according to the specified
+     * schedule.
+     */
+//    @TimerTrigger(name = "ReportingBatchTrigger", schedule = "0 */1 * * * *") String timerInfo,
+
+    @FunctionName("ReportingBatchFunction")
+    public void run(
+            @TimerTrigger(name = "ReportingBatchTrigger", schedule = "*/5 * * * * *") String timerInfo,
```
Comment (jacopocarlini) [inline reporting-batch/src/main/java/it/gov/pagopa/reporting/RetrieveOrganizations.java:35]: value of `schedule` is valid also for production?

File: reporting-batch/src/main/java/it/gov/pagopa/reporting/entity/OrganizationEntity.java (hunk 4 above)
Comment (pasqualespica) [reply reporting-batch/src/main/java/it/gov/pagopa/reporting/entity/OrganizationEntity.java:26]: right

Comment (pasqualespica) [reply reporting-batch/src/main/java/it/gov/pagopa/reporting/entity/OrganizationEntity.java:26]: ok, i'll add comment

File: reporting-batch/src/main/java/it/gov/pagopa/reporting/service/FlowsService.java (hunk 2 above)
Comment (pasqualespica) [reply reporting-batch/src/main/java/it/gov/pagopa/reporting/service/FlowsService.java:43]: you're right, in fact this is useless code that is only for local development for azurite

**Review by aomegax — APPROVED** [review]
🥇

**Pull request merged**